vxfreeswitch/tests/test_voice.py

```diff
@@ -444,3 +462,102 @@ def test_reply_to_client_that_has_hung_up(self):
         self.assertEqual(nack['user_message_id'], msg['message_id'])
         self.assertEqual(nack['nack_reason'],
                          "Client u'TESTTEST' no longer connected")
+
+
+class RecordingServer(Protocol):
+    def connectionMade(self):
+        self.factory.clients.append(self)
+
+    def _send_event(self, content):
+        self.transport.write(
+            'Content-Length: %s\n' % len(content) +
+            'Content-Type: text/event-plain\n\n' +
+            content)
+
+    def dataReceived(self, line):
+        self.factory.data.append(line)
```

Comment:
I think you can feed this data into an ``EslParser`` (see higher up in this file) and have that parse commands for you. The benefits of that are two fold:
1) It will correctly handle data that is split across multiple calls to ``dataReceived``
2) It will allow us to write better asserts in tests (since we will have command objects, rather than raw strings).